dipy/workflows/align.py

```diff
@@ -698,3 +699,188 @@ def run(self, static_image_file, moving_image_files, affine_matrix_file,
+        if metric not in ['ssd', 'cc', 'em']:
+            raise ValueError("Invalid similarity metric: Please"
+                             " provide a valid metric.")
+
+        # Init parameter if they are not setup
+        init_param = {'ssd': {'mopt_smooth': 4.0,
+                              'mopt_inner_iter': 10,
+                              'mopt_step_type': 'demons'
+                              },
+                      'em': {'mopt_smooth': 1.0,
+                             'mopt_inner_iter': 5,
+                             'mopt_step_type': 'gauss_newton'
+                             }
+                      }
+        mopt_smooth = mopt_smooth or init_param[metric]['mopt_smooth']
+        mopt_inner_iter = mopt_inner_iter or  \
+            init_param[metric]['mopt_inner_iter']
+        mopt_step_type = mopt_step_type or \
+            init_param[metric]['mopt_step_type']
+
+        for static_file, moving_file, in_affine, \
+                warped_file, inv_static_file, displ_file in io_it:
+
+            # Loading the image data from the input files into object.
+            static_image, static_grid2world = load_nifti(static_file)
+            moving_image, moving_grid2world = load_nifti(moving_file)
+
+            # Sanity check for the input image dimensions.
+            ImageRegistrationFlow.check_dimensions(static_image, moving_image)
+
+            # Loading the affine matrix.
+            affine_matrix = np.loadtxt(in_affine)
+
+            l_metric = {"ssd": SSDMetric(static_image.ndim,
+                                         smooth=mopt_smooth,
+                                         inner_iter=mopt_inner_iter,
+                                         step_type=mopt_step_type
+                                         ),
+                        "cc": CCMetric(static_image.ndim,
+                                       sigma_diff=mopt_sigma_diff,
+                                       radius=mopt_radius),
+                        "em": EMMetric(static_image.ndim,
+                                       smooth=mopt_smooth,
+                                       inner_iter=mopt_inner_iter,
+                                       step_type=mopt_step_type,
+                                       q_levels=mopt_q_levels,
+                                       double_gradient=mopt_double_gradient)
+                        }
+
+            current_metric = l_metric.get(metric.lower())
+
+            sdr = SymmetricDiffeomorphicRegistration(current_metric,
+                                                     level_iters)
+
+            mapping = sdr.optimize(static_image, moving_image,
+                                   static_grid2world, moving_grid2world,
+                                   affine_matrix)
+
+            warped_moving = mapping.transform(moving_image)
+
+            # Saving the warped moving file and the alignment matrix.
+            save_nifti(warped_file, warped_moving, static_grid2world)
```

Comment:
Should we also save out the transformation? That is, the X by Y by Z by 3 values of the diffeomorphism? We could then apply this transformation to other files, which may be useful. 

We have an implementation of something like that here: 

https://github.com/yeatmanlab/pyAFQ/blob/master/AFQ/registration.py#L138
https://github.com/yeatmanlab/pyAFQ/blob/master/AFQ/registration.py#L153

Reply:
sounds good, will do

Reply:
Great! We'll want to eventually follow up with an API to apply these transforms to input files. But that's probably for another release cycle...

Reply:
Yes, I think of changing [ApplyAffineFlow](https://github.com/nipy/dipy/blob/master/dipy/workflows/align.py#L645) to make it more generic. I want to call it `ApplyTransformFlow` with an option `--affine` or `--diffeormorphic`. 

I will see if I have time to do that today. Otherwise, next release....